test/functions/math.test.cpp

```diff
@@ -61,3 +62,27 @@ TEST_CASE("Randomised test for big_pow10", "[functions][math][big_pow10][random]
         REQUIRE(big_num.to_string() == "1" + std::string(rand_exponent, '0'));
     }
 }
+
+TEST_CASE("Pow with BigInt", "[conversion][string]") {
+    BigInt a = 12;
+
+    REQUIRE(pow(a, (long long)32) == "34182189187166852111368841966125056");
+    a = 32;
+    REQUIRE(pow(a, (long long)12) == "1152921504606846976");
+    a = 50;
+    REQUIRE(pow(a, (long long)23) == "1192092895507812500000000000000000000000");
+}
+
+TEST_CASE("Pow with long long", "[conversion][string]") {
+    REQUIRE(pow((long long)20, (long long)40) == "10995116277760000000000000000000000000000000000000000");
+    REQUIRE(pow((long long)30, (long long)45) == "2954312706550833698643000000000000000000000000000000000000000000000");
+    REQUIRE(pow((long long)45, (long long)30) == "39479842665806602234295041487552225589752197265625");
+    REQUIRE(pow((long long)40, (long long)20) == "109951162777600000000000000000000");
```

Comment:
No need to explicitly cast to `long long`.

Reply:
It doesn't compile otherwise.

Reply:
I think only `pow(long long, long long)` causes problems. Try casting just the `base` as `long long` if that's the case.

Reply:
The workaround would be to write `ll` at the end.

Reply:
Use uppercase `LL` though since its easier to differentiate from `11`.